### PR TITLE
MGMT-8987: Update master/worker count validation message

### PR DESCRIPTION
**Pull request by flaper87**
Signed-off-by: Flavio Percoco <flavio@redhat.com>

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

Manual test was done using @jkilzi beaker machine

## Checklist

- [x] Title and description added to both, commit and PR.
- [ x Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by jkilzi** [comment]
LGTM

**Comment by gamli75** [comment]
@flaper87 please fix the following test: https://github.com/openshift/assisted-service/blob/5c9745d2283726f48d2d25c859b2ec5730d863f9/internal/operators/odf/validation_test.go#L304

File: internal/cluster/validator.go [hunk 1]
```diff
@@ -315,8 +315,7 @@ func (v *clusterValidator) sufficientMastersCount(c *clusterPreprocessContext) (
 	case ValidationSuccess:
 		message = "The cluster has a sufficient number of master candidates."
 	case ValidationFailure:
-		message = fmt.Sprintf("Clusters must have exactly %d dedicated masters and if workers are added, there should be at least 2 workers. Please check your configuration and add or remove hosts as to meet the above requirement.",
-			common.MinMasterHostsNeededForInstallation)
+		message = fmt.Sprintf("Clusters at most %d dedicated masters. Please check your configuration and add or remove hosts as needed to meet the requirement.", common.MinMasterHostsNeededForInstallation)
```
Comment (karmab) [inline internal/cluster/validator.go:318]: -> Clusters must have at most

Comment (flaper87) [inline internal/cluster/validator.go:318]:
```suggestion
		message = fmt.Sprintf("Clusters most have, at most, %d dedicated masters. Please check your configuration and add or remove hosts as needed to meet the requirement.", common.MinMasterHostsNeededForInstallation)
```

File: internal/operators/odf/validation_test.go [hunk 2]
```diff
@@ -328,7 +328,7 @@ var _ = Describe("Ocs Operator use-cases", func() {
 				clust.AllHostsAreReadyToInstall:           {status: clust.ValidationSuccess, messagePattern: "All hosts in the cluster are ready to install"},
 				clust.IsDNSDomainDefined:                  {status: clust.ValidationSuccess, messagePattern: "The base domain is defined"},
 				clust.IsPullSecretSet:                     {status: clust.ValidationSuccess, messagePattern: "The pull secret is set"},
-				clust.SufficientMastersCount:              {status: clust.ValidationFailure, messagePattern: "Clusters must have exactly 3 dedicated masters"},
+				clust.SufficientMastersCount:              {status: clust.ValidationFailure, messagePattern: "Clusters most have, at most, 3 dedicated masters."},
```
Comment (gamli75) [inline internal/operators/odf/validation_test.go:331]: Same as Avishay's remark

Comment (flaper87) [inline internal/operators/odf/validation_test.go:331]:
```suggestion
				clust.SufficientMastersCount:              {status: clust.ValidationFailure, messagePattern: "Clusters must have at most 3 dedicated masters"},
```

**Review by avishayt — APPROVED** [review]

**Comment by flaper87** [comment]
retesting since I ran this test locally and it passes: 

/test edge-unit-test

**Pull request merged**